docs/docs/reference/Vectors.md

```diff
@@ -173,11 +173,29 @@ Every "`*_attribute`" parameter should refer to an attribute in the [`PARAMS`](/
 
 *   `[ AS {score field name | $score_field_name_attribute}]` - An optional part for specifying a score field name, for later sorting by the similarity score. By default the score field name is "`__{vector field}_score`" and it can be used for sorting without using `AS {score field name}` in the query.
 
-## Specific runtime attributes per algorithm
+### Hybrid vector similarity search
 
+We refer vector similarity queries of the form `{primary filter query}=>[{vector similarity query}]` as *hybrid queries*. RediSearch has an internal mechanism for optimizing the computation of such queries. There are 2 modes in which hybrid queries are executed: 
+1. **Batches mode** - an iterative procedure, where in each step a batch of the top results that correspond to the `{vector similarity query}` part of the query are obtained from the vector index. Then, every result (which represents a certain document in the index) is evaluated, and returned only if the rest of the document's fields pass the `{primary filter query}` part of the query. If the accumulated number of results hadn't reached the requested `k` results, this procedure continues and another batch of the *next best* results is obtained from the vector index in the next step. The procedure terminates when `k` documents that pass the `{primary filter query}` are collected (or until every vector in the index was obtained and processed).
```

Comment:
In this mode, a batch of the high-scoring documents from the vector index are retrieved. These documents are returned ONLY if `{primary filter query}` is satisfied. (e.g. the document contains a similar vector and meets the filter criteria)
The procedure terminates when `k` documents that pass the `{primary filter query}` are returned or after every vector in the index was obtained and processed